Sources/AWSLambdaRuntimeCore/LambdaRunner.swift

```diff
@@ -57,6 +58,7 @@ extension Lambda {
                 let context = Context(logger: logger, eventLoop: self.eventLoop, invocation: invocation)
                 logger.debug("sending invocation to lambda handler \(handler)")
                 return handler.handle(context: context, event: event)
+                    .hop(to: self.eventLoop)
```

Comment:
👀

Reply:
@tomerd The only situation in which we would need to `hop` is if developers bring their own `EventLoopGroup`. Is that something that is likely?

Reply:
this can happen is if they use some library (lets say a DB client) that manages it's own ELG for whatever reason and they return a future that originated from that ELG. while this is not a pattern we encourage, we also can't stop anyone from doing it so to be safe we should hop imo

Reply:
Vapor doesn't do a `hop` in its routes and quite a number of people hit that issue at some point. After that they know its an issue and don't do it again. There is an argument for education here instead of protecting against. But at the same time a `hop` onto the same `EventLoop` is hardly a performance hit.

Reply:
Just as a reference: Where ever NIO takes a future (not that many APIs outside of bootstraps), it'll `hop` it to the correct event loop.